pkg/controllers/pushsecret/pushsecret_controller.go

```diff
@@ -246,9 +247,18 @@ func (r *Reconciler) PushSecretToProviders(ctx context.Context, stores map[esapi
 			return out, fmt.Errorf("could not get secrets client for store %v: %w", store.GetName(), err)
 		}
 		for _, ref := range ps.Spec.Data {
-			secretValue, ok := secret.Data[ref.Match.SecretKey]
-			if !ok {
-				return out, fmt.Errorf("secret key %v does not exist", ref.Match.SecretKey)
+			var secretValue []byte
+			if ref.Match.SecretKey != "" {
+				var ok bool
+				secretValue, ok = secret.Data[ref.Match.SecretKey]
+				if !ok {
+					return out, fmt.Errorf("secret key %v does not exist", ref.Match.SecretKey)
+				}
+			} else {
+				secretValue, err = json.Marshal(secret.Data)
+				if err != nil {
+					return out, fmt.Errorf("error marshaling vault secret: %w", err)
+				}
 			}
```

Comment:
Instead of marshaling the secret data we should change the `PushSecret()` interface. Then the provider can decide on how to handle that data.
Reason: 
- the unneccessary marshal/unmarshal is error prone: binary data isn't well supported (needs encoding, e.g. base64) and then you can't differentiate the data: is it a string, binary-base64 encoded, a jwt :shrug: ?
- the provider doesn't know about the `ref.Match`, hence can not differentiate whether this belongs to a `match.seretKey` or this is the whole secret.

Sorry for the late reply, we're under water with PR reviews :disappointed:
Do you have to to finish this up? If not we can take it over

Reply:
Hi @moolen, unfortunately I will not have time to refactor the `PushSecret()` interface, feel free to take over.

Good luck with the other PR reviews!